lab_1/main.py

```diff
@@ -86,10 +146,41 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
-
-
-def detect_language_advanced(unknown_profile: dict, profiles: list, languages: list, top_n: int) -> str or None:
+    if isinstance(unknown_profile, dict) and isinstance(profile_to_compare, dict)\
+        and isinstance(top_n, int):
+        lang_adv = {}
+        cross = []
+        lang_adv['name'] = profile_to_compare['name']
+        for i in get_top_n_words(profile_to_compare['freq'], top_n):
+            if i in get_top_n_words(unknown_profile['freq'], top_n):
+                cross.append(i)
+            else:
+                continue
```

Comment:
need this?

Reply:
I thought else was necessary (like with the float) (I don't actually know why I thought so, actually) but now it was removed